core/src/dbs/capabilities.rs

```diff
@@ -237,14 +354,22 @@ pub struct Capabilities {
 	deny_funcs: Arc<Targets<FuncTarget>>,
 	allow_net: Arc<Targets<NetTarget>>,
 	deny_net: Arc<Targets<NetTarget>>,
+
+	// Server capabilities
+	// These capabilities limit what the SurrealDB server allows
+	// Server capabilities are not exposed through the library
+	allow_rpc: Arc<Targets<MethodTarget>>,
```

Comment:
There is a lot of arc here. And, as an optimisation,  I am wondering if Targets should not be cloneable. And rather to always encapsulate it to an Arc, having only the HashSet on Some being Arced.

Targets would then be like that:

```pub enum Targets<T: Hash + Eq + PartialEq> {
	None,
	Some(Arc<HashSet<T>>),
	All,
}
```

Reply:
I see your point! Although I absolutely agree, I think we should dedicate a PR to optimizing this rather than doing it on this PR, which just adds two new capabilities to the existing system without changing it. Would you be okay with resolving this comment so we can merge this PR, @emmanuel-keller?